clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1169,16 +1188,16 @@ class SyclKernelBodyCreator
   void handleSpecialType(FieldDecl *FD, QualType Ty) {
     const auto *RecordDecl = Ty->getAsCXXRecordDecl();
     // Perform initialization only if it is field of kernel object
-    if (MemberExprBases.size() == 1) {
-      InitializedEntity Entity =
-          InitializedEntity::InitializeMember(FD, &VarEntity);
-      // Initialize with the default constructor.
-      InitializationKind InitKind =
-          InitializationKind::CreateDefault(SourceLocation());
-      InitializationSequence InitSeq(SemaRef, Entity, InitKind, None);
-      ExprResult MemberInit = InitSeq.Perform(SemaRef, Entity, InitKind, None);
-      InitExprs.push_back(MemberInit.get());
-    }
+    // if (MemberExprBases.size() == 1) {
+    InitializedEntity Entity =
+        InitializedEntity::InitializeMember(FD, &VarEntity);
```

Comment:
I'm a bit concerned about this place since `VarEntity` is some stuff created for kernel object clone variable. I think we need to create corresponding `InitializedEntity` for each structure when we enter it.